### ISSUE TITLE
Some weird bugs occurred when I'm trying to build a bipartite batch following the tutorial.

### ISSUE THREAD

**Issue by Emiyalzn** [pyg-team/pytorch_geometric#3137]
I'm using the latest pyg and trying to build a bipartite batch for training.
The code is listed as following:
```python
import torch
from torch_geometric.data import Data, DataLoader

class BipartiteData(Data):
    def __init__(self, edge_index, x_src, x_dst):
        super(BipartiteData, self).__init__()
        self.edge_index = edge_index
        self.x_src = x_src
        self.x_dst = x_dst
        
    def __inc__(self, key, value):
        if key == 'edge_index':
            return torch.tensor([[self.x_src.size(0)], [self.x_dst.size(0)]])
        else:
            return super(BipartiteData, self).__inc__(key, value)


x_src = torch.randn(100, 32)
x_dst = torch.randn(50, 32)
row = torch.randint(0, x_src.size(0), (200, ), dtype=torch.long)
col = torch.randint(0, x_dst.size(0), (200, ), dtype=torch.long)
edge_index = torch.stack([row, col], dim=0)

data = BipartiteData(x_src=x_src, x_dst=x_dst, edge_index=edge_index)

loader = DataLoader([data] * 8, batch_size=4, follow_batch=['x_src', 'x_dst'])
for batch in loader:
    print(batch)
```
I think there is nothing wrong with my code, but when I called `python test.py`, the traceback is as following:
```C
/opt/anaconda3/lib/python3.8/site-packages/torch_geometric/deprecation.py:13: UserWarning: 'data.DataLoader' is deprecated, use 'loader.DataLoader' instead
  warnings.warn(out)
Traceback (most recent call last):
  File "test.py", line 27, in <module>
    for batch in loader:
  File "/opt/anaconda3/lib/python3.8/site-packages/torch/utils/data/dataloader.py", line 435, in __next__
    data = self._next_data()
  File "/opt/anaconda3/lib/python3.8/site-packages/torch/utils/data/dataloader.py", line 475, in _next_data
    data = self._dataset_fetcher.fetch(index)  # may raise StopIteration
  File "/opt/anaconda3/lib/python3.8/site-packages/torch/utils/data/_utils/fetch.py", line 47, in fetch
    return self.collate_fn(data)
  File "/opt/anaconda3/lib/python3.8/site-packages/torch_geometric/loader/dataloader.py", line 39, in __call__
    return self.collate(batch)
  File "/opt/anaconda3/lib/python3.8/site-packages/torch_geometric/loader/dataloader.py", line 19, in collate
    return Batch.from_data_list(batch, self.follow_batch,
  File "/opt/anaconda3/lib/python3.8/site-packages/torch_geometric/data/batch.py", line 45, in from_data_list
    batch, slice_dict, inc_dict = collate(
  File "/opt/anaconda3/lib/python3.8/site-packages/torch_geometric/data/collate.py", line 29, in collate
    out = cls(_base_cls=data_list[0].__class__)  # Dynamic inheritance.
  File "/opt/anaconda3/lib/python3.8/site-packages/torch_geometric/data/batch.py", line 22, in __call__
    return super(DynamicInheritance, new_cls).__call__(*args, **kwargs)
TypeError: __init__() missing 3 required positional arguments: 'edge_index', 'x_src', and 'x_dst'
```
I've tried many times and couldn't fix this bug, I'll really appreciate it if you could give me any help.

**Comment by rusty1s** [comment]
Thanks for letting me know. This is indeed a bug and I fixed it in master. In the meantime, you can get around by making input arguments to `BipartiteData` optional:
```python
class BipartiteData(Data):
    def __init__(self, edge_index=None, x_src=None, x_dst=None):
```

**Comment by Emiyalzn** [comment]
Thank you so much! I'll create this issue now.

**Comment by rusty1s** [comment]
I just released `torch-geometric==2.0.1` that fixes this issue :)
